src/ansi-c/c_typecheck_code.cpp

```diff
@@ -481,9 +481,9 @@ void c_typecheck_baset::typecheck_for(codet &code)
     }
 
     code_block.reserve_operands(2);
-    code_block.move_to_operands(code.op0());
+    code_block.move(to_code(code.op0()));
     code.op0().make_nil();
-    code_block.move_to_operands(code);
+    code_block.move(code);
     code.swap(code_block);
```

Comment:
This code is a bit strange: the `code.op0().make_nil();` appears to have lasting effect.

Reply:
yes, see comment above the if(...) -- this is deliberate.